src/protocol/resp/src/parse/error.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023 Pelikan Foundation LLC.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+use std::borrow::Cow;
+use std::fmt;
+
+use bstr::BStr;
+use thiserror::Error;
+
+#[derive(Error)]
+pub enum ParseError<'a> {
+    #[error("ran out of data unexpectedly")]
```

Comment:
The "unexpectedly" is not exactly accurate, since a command going through TCP can be delivered partially for a number of reasons, and both ends should see this as a somewhat normal condition. (This is going to be more common if the value is really large, and memcached/pelikan-legacy both have special handling of a partial command with large value, memcached also tries to avoid double copying. OTOH Redis does not iirc. But I digressed...)

I think "not enough data, parsing incomplete" or something like that would be a better message.